### PR TITLE
Add admin role level

### PR DESCRIPTION
**Pull request by Draganovic**
Create admin dashboard namespace
Only admin can add a book

### CODE REVIEW

**Comment by Draganovic** [comment]
Ready to merge

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -9,8 +9,12 @@ def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
-  def authorize
+  def rating_authorize
```
Comment (AllPurposeName) [inline app/controllers/application_controller.rb:12]: why this name change?

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -4,8 +4,28 @@ def index
...
+  def new
+    @book = Book.new
+  end
+
+  def create
+    @book = Book.new(book_params)
+    if @book.save
+      redirect_to books_path
+    else
+      flash.now[:error] = "Please enter a title an author and image for the book."
```
Comment (AllPurposeName) [inline app/controllers/books_controller.rb:16]: What does `flash.now[:error] =` do differently from `render :new, flash: { error: "error goes here" }`

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -9,7 +9,11 @@ def create
         # @user = User.find(session[])
         # set the current session's user_id tp this @user
         session[:user_id] = @user.id
-        redirect_to user_path(@user)
+        if @user.admin?
```
Comment (AllPurposeName) [inline app/controllers/sessions_controller.rb:12]: How could this logic be extracted from the controller?

File: spec/features/admin_can_add_a_book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
```
Comment (AllPurposeName) [inline spec/features/admin_can_add_a_book_spec.rb:1]: This test is solid. I would, however, like to see some kind of proof that a User can NOT add a book.

**Comment by AllPurposeName** [comment]
Small commits like this are good, small PRs like this are unrealistic. If you're tasked with adding admin functionality, you are going to need a lot more than just an admin adding a book. I'd like to see all the base admin features (CRUD) in one PR.

**Comment by AllPurposeName** [comment]
Only seeing one spec added here. There weren't any controller or model tests you could add? If you need a template for a controller test, feel free to ping me. I learned it from copy/pasta, so no judgement there!

**Pull request merged**